src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/WorkspaceProjectContextHost.WorkspaceProjectContextHostInstance.cs

```diff
@@ -30,11 +28,11 @@ internal partial class WorkspaceProjectContextHostInstance : OnceInitializedOnce
             private readonly ExportFactory<IApplyChangesToWorkspaceContext> _applyChangesToWorkspaceContextFactory;
             private readonly IDataProgressTrackerService _dataProgressTrackerService;
 
-            private IDataProgressTrackerServiceRegistration _evaluationProgressRegistration;
-            private IDataProgressTrackerServiceRegistration _projectBuildProgressRegistration;
-            private DisposableBag _disposables;
-            private IWorkspaceProjectContextAccessor _contextAccessor;
-            private ExportLifetimeContext<IApplyChangesToWorkspaceContext> _applyChangesToWorkspaceContext;
+            private IDataProgressTrackerServiceRegistration? _evaluationProgressRegistration;
+            private IDataProgressTrackerServiceRegistration? _projectBuildProgressRegistration;
+            private DisposableBag? _disposables;
+            private IWorkspaceProjectContextAccessor? _contextAccessor;
```

Comment:
Shouldn't this be non-null?

Reply:
It's not assigned in the constructor. If we want it to be non-null, we either need a non-null placeholder (as in the case of empty collections) or to `#pragma warning disable` the ctor, which doesn't feel great. I don't think there's a good placeholder here.

This happens a fair bit in this repo, where things are initialised after construction.